prov/sm2/src/sm2.h

```diff
@@ -304,6 +316,16 @@ static inline size_t sm2_pop_xfer_entry(struct sm2_ep *ep,
 	return FI_SUCCESS;
 }
 
+static inline bool sm2_proto_imm_send_comp(uint16_t proto)
+{
+	switch (proto) {
```

Comment:
I think this could be written better as:
```
if (inject) {
   return true
}
return false
```

My gut instinct is telling me switch statements have more overhead than if, but I would need to verify with godbolt (might be wrong).

Reply:
if statement: https://godbolt.org/z/j8aW6novo